flatpak_builder_lint/appstream.py

```diff
@@ -74,3 +74,21 @@ def has_manifest_key(path: str) -> bool:
         if key.attrib.get("key") == "flathub::manifest":
             return True
     return False
+
+
+def has_icon_key(path: str) -> bool:
+    return bool(components(path)[0].xpath("icon"))
+
+
+def is_icon_remote(path: str) -> bool:
+    for icon in parse_xml(path).findall("component/icon"):
+        if "remote" in str(icon.attrib):
+            return True
```

Comment:
Needs to be modified to check that remote icons have texts starting with https://dl.flathub.org/media

If not it's considered broken.